plugins/com.aptana.js.core/src/com/aptana/js/internal/core/node/NodePackageManager.java

```diff
@@ -489,16 +489,23 @@ public IPath getModulesPath(String packageName, boolean isGlobal, String... args
 		{
 			CollectionsUtil.addToList(processArgs, args);
 		}
-
+		
 		IStatus status = runInBackground(CollectionsUtil.toArray(processArgs));
-		if (!status.isOK())
+		IPath moduleIPath = null;
+		boolean isPathExists = false;
+		String message = status.getMessage();
+		if (!StringUtil.isEmpty(message))
+		{
+			String[] lines = message.split("\n"); //$NON-NLS-1$
+			moduleIPath = Path.fromOSString(lines[lines.length - 1]);
+			isPathExists = moduleIPath.toFile().exists();
+		}
+		if (!status.isOK() && !isPathExists) //some cases status.isNotOK even though it finds valid path
 		{
 			throw new CoreException(new Status(IStatus.ERROR, JSCorePlugin.PLUGIN_ID, MessageFormat.format(
 					Messages.NodePackageManager_FailedListPackageError, packageName)));
```

Comment:
Since we already throwing an exception only If status is not Ok and Path doesn't exist, and the caller anyway might be catching. Still do you think we need to log it here ?